### PR TITLE
chore: remove task categories

### PR DESCRIPTION
**Pull request by Chriscbr**
Since the start menu was removed in v0.24.6, task categories are no longer used, and can be removed.

I'm open to leaving them in if we think they can be used in other ways in the future (as metadata or otherwise).

BREAKING CHANGE: Task categories have been removed.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by eladb** [comment]
Lol I was just thinking we should get rid of these!

File: src/node-package.ts [hunk 1]
```diff
@@ -686,7 +686,7 @@ export class NodePackage extends Component {
   }
 
   private determineVersion(currVersion?: string) {
-    if (!this.isRelaseBuild) {
```
Comment (eladb) [inline src/node-package.ts:689]: ☹️

File: src/tasks/model.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export interface TaskCommonOptions {
    * Category for start menu.
    *
    * @default TaskCategory.MISC
+   * @deprecated - start menu has been removed (use `projen --help`)
```
Comment (eladb) [inline src/tasks/model.ts:27]: Just remove this I think. I doubt it will break anyone

**Review by eladb — CHANGES_REQUESTED** [review]

File: src/node-package.ts (hunk 1 above)
Comment (Chriscbr) [reply src/node-package.ts:689]: do you want this omitted?

**Review by eladb — APPROVED** [review]

**Pull request merged**